### PR TITLE
feat: Updating Antrea version to v0.3.0

### PR DESCRIPTION
**Pull request by reachjainrahul**
Release note covers the changes
https://github.com/vmware-tanzu/antrea/releases/tag/v0.3.0

**Reason for Change**:

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by jackfrancis** [comment]
Thanks @reachjainrahul! You'll need to run `make generate` (and commit the changes) to make the generated code validator happy.

**Comment by reachjainrahul** [comment]
> Thanks @reachjainrahul! You'll need to run `make generate` (and commit the changes) to make the generated code validator happy.

Done

**Comment by reachjainrahul** [comment]
@jackfrancis Could you please re-run e2e tests on the updated change. Thanks!

File: parts/k8s/addons/antrea.yaml [hunk 1]
```diff
@@ -332,9 +372,26 @@ spec:
         key: node-role.kubernetes.io/master
       volumes:
       - configMap:
-          name: antrea-config-48gttf992h
+          name: antrea-config-476k567258
         name: antrea-config
 ---
+---
```
Comment (jackfrancis) [inline parts/k8s/addons/antrea.yaml]: Cluster deployments are failing on this PR against this test config:

https://github.com/Azure/aks-engine/blob/master/test/e2e/test_cluster_configs/network_policy/antrea.json

There is an extra `---` above, perhaps that's failing to be parsed by `kube-addon-manager`.

Comment (reachjainrahul) [reply parts/k8s/addons/antrea.yaml]: Verified the latest change locally by running e2e tests and it passed. Please re-run e2e test for checkin. Thanks!

**Comment by reachjainrahul** [comment]
Hi @jackfrancis 

I see E2E tests failing with K8Version 1.15 and 1.16. And from the logs, its failing at Windows Port Forwarding test
By("Creating a Windows IIS deployment")

I see we are deploying both linux and windows VM in the same topology. However, Antrea works only for Linux VM. Shouldnt the test infra account for this?

Thanks
Rahul

**Comment by jackfrancis** [comment]
@reachjainrahul We don't actually test antrea in github PR E2E CI, we have an internal jenkins instance that is able to do this. That signal was 🍏 so we're on the merge train. :)

**Pull request merged**